url_finder/src/api/find_retri_by_client.rs

```diff
@@ -0,0 +1,126 @@
+use std::{sync::Arc, time::Duration};
+
+use axum::{debug_handler, extract::State, Json};
+use axum_extra::extract::WithRejection;
+use color_eyre::Result;
+use common::api_response::{internal_server_error, ok_response, ApiResponse, ErrorResponse};
+use serde::{Deserialize, Serialize};
+use tokio::time::timeout;
+use tracing::debug;
+use utoipa::ToSchema;
+
+use crate::{deal_service, provider_endpoints, url_tester, AppState};
+
+use super::ResultCode;
+
+#[derive(Deserialize, ToSchema)]
+pub struct FindRetriByClientAndSpInput {
+    pub client: String,
+    pub provider: String,
+}
+
+#[derive(Serialize, ToSchema)]
+pub struct FindRetriByClientAndSpResponse {
+    pub result: ResultCode,
+    pub retrievability_percent: f64,
+}
+
+const RETRIEVABILITY_TIMEOUT_SEC: u64 = 2 * 60; // 2 min
+
+/// Find retrivabiliy of urls for a given SP and Client address
+#[utoipa::path(
+    post,
+    path = "/url/retri",
```

Comment:
```suggestion
    path = "/url/retrievability",
```

Let's use the full word in public interfaces